.github/workflows/ci.yml

```diff
@@ -35,4 +35,4 @@ jobs:
       - name: Check licenses
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: make license-check
+        run: go mod vendor && make license-check
```

Comment:
thanks for your suggestion @pregnor 

https://github.com/goph/licensei has two modes of validating the license, if the dependency is vendored the LICENSE file is retrieved from the local vendored location and another where the LICENSE file is retrieved from github.com repo (if a github repo can be determined for a dependency)

The second mode is prone to false positive when the github repo cannot be determined for a dependency. I discovered that while running the license-check target after the deps updates here. To avoid this, I proposed we vendor the deps before running the license checker.

I extracted the vendoring in a separate github action task and added more comments to reflect this.
lmk

Reply:
Thank you for the explanation, it makes sense, I didn't realize this can be a solution to the licensei pain we experience, good idea.